xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -1161,22 +1160,31 @@ static void DisplayReconfigured(CGDirectDisplayID display,
   if (m_appWindow)
   {
     dispatch_sync(dispatch_get_main_queue(), ^{
-      OSXGLView* contentView = m_appWindow.contentView;
-      cglcontex = contentView.getGLContext.CGLContextObj;
+      cglcontex = [(OSXGLView*)m_appWindow.contentView getGLContextObj];
     });
   }
 
   return cglcontex;
 }
 
+CGraphicContext& CWinSystemOSX::GetGfxContext() const
+{
+  if (m_glView)
+  {
+    dispatch_sync(dispatch_get_main_queue(), ^{
```

Comment:
Note to self: this deadlocks if called from the main thread (the windowing code). Happens on display reconfigure and a few other calls.
Needs further work (maybe accessing the m_gfxctx directly in such cases/this class).
If anyone has further ideas they are appreciated.
If anyone has any ideas